pkg/pipeline/extract/timebased/filters.go

```diff
@@ -149,10 +150,21 @@ func (fs *FilterStruct) CreateGenericMap() []config.GenericMap {
 			"index_key":        fs.Rule.IndexKey,
 			"operation":        fs.Rule.OperationType,
 			"operation_key":    fs.Rule.OperationKey,
-			"key":              result.key,
+			"key":              result.values,
 			"operation_result": result.operationResult,
 		}
-		t[fs.Rule.IndexKey] = result.key
+
+		// append operation key and result as key / value
+		t[fs.Rule.OperationKey] = result.operationResult
```

Comment:
I feel more user friendly to output the `<operation_key>: <operation_result>` as output like `"Bytes":500` especially when [reusing these in prometheus encode](https://github.com/netobserv/network-observability-operator/pull/391/files#diff-2ec52703296accf2f2ab72dea73684fe5433df463b13cd15765e856a9edbe80a)

Full output example:
```
{
   "key":"10.0.0.2,11.0.0.1,0",
   "name":"BotK_SrcDst_Bytes",
   "operation":"avg",
   "operation_key":"Bytes",
   "operation_result":500,
   "Bytes":500,
   "index_key":"SrcAddr,DstAddr,Direction",
   "SrcAddr":"10.0.0.2",
   "DstAddr":"11.0.0.1",
   "Direction":"0"
}
```

Following encode example:
```
encode:
  type: prom
  prom:
    metrics:
      - name: top_pods_egress_bytes_total
        type: counter
        valuekey: Bytes
        filters:
          - key: FlowDirection
            value: "1"
        labels:
          - SrcK8S_Name
          - SrcK8S_Namespace
          - SrcK8S_Type
          - SrcK8S_HostName
```

Maybe also we can remove `operation_key` & `operation_result` (unless we want to keep compatibility with previous versions) ?

Reply:
Similarly `key` could be removed from the JSON

Reply:
I am not aware of who (if anybody) is using the timebased TopK feature. The specified fields were included because it was thought that they might be useful for the user, but only user feedback can give us a real indication. If you find your proposed fields useful for what you are doing, then this is the first concrete user feedback we have, and you should add whatever fields you find useful.

Reply:
I think the `operation_key` should be kept to help the user remember what action was performed to produce the metric. By the time the user sees the metric, he may no longer be aware of the aggregation that was performed to produce it, so some indication in the flow log might be helpful.